src/python/pants/goal/run_tracker.py

```diff
@@ -179,6 +179,7 @@ def start(self, all_options: Options, run_start_time: float) -> None:
         self._main_root_workunit.start(run_start_time)
         self.report.start_workunit(self._main_root_workunit)
 
+        self.run_info.add_info("goals", all_options.goals)
```

Comment:
On the line below, we call `tuple`. Do you know why and if we need it here?

Reply:
`all_options.goals` returns a list, that `tuple` call just converts it into a tuple. I don't think there's any principled reason why the type of this value needs to be one or the other. It happens that the toolchain API that currently consumes `self._v2_goal_rule_names` converts it back into a `list`, but I don't think that's a meaningful choice either.